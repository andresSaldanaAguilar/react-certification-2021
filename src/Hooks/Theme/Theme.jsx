import React, { createContext, useContext, useState } from 'react';

const ThemeStateContext = createContext();
const ThemeDispatchContext = createContext();

function ThemeProvider({ children }) {
  const [theme, setTheme] = useState('light');
  return (
    <ThemeStateContext.Provider value={theme}>
      <ThemeDispatchContext.Provider value={setTheme}>
        {children}
      </ThemeDispatchContext.Provider>
    </ThemeStateContext.Provider>
  );
}
function useThemeState() {
  const context = useContext(ThemeStateContext);
  if (context === undefined) {
    throw new Error('useThemeState must be used within a ThemeProvider');
  }
  return context;
}
function useThemeDispatch() {
  const context = useContext(ThemeDispatchContext);
  if (context === undefined) {
    throw new Error('useThemeDispatch must be used within a ThemeProvider');
  }
  return context;
}
export { ThemeProvider, useThemeState, useThemeDispatch };
