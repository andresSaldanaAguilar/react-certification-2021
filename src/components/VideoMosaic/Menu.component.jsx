import React from 'react';
import { IconButton, Typography, Button } from '@material-ui/core';
import YouTubeIcon from '@material-ui/icons/YouTube';
import SearchIcon from '@material-ui/icons/Search';
import {
  CustomAppBar,
  CustomInputBase,
  CustomToolbar,
  SearchContainer,
  SearchIconContainer,
  ThemeSwitch,
  ToolbarSection,
} from './Menu.styled';

function Menu() {
  const [checked, setChecked] = React.useState({
    checked: false,
  });

  const handleChange = () => {
    setChecked(!checked);
  };

  return (
    <>
      <CustomAppBar position="static">
        <CustomToolbar>
          <ToolbarSection>
            <IconButton className="title" edge="start" color="inherit" aria-label="menu">
              <YouTubeIcon />
              <Typography variant="body1" noWrap>
                MyTube
              </Typography>
            </IconButton>
            <SearchContainer>
              <SearchIconContainer>
                <SearchIcon />
              </SearchIconContainer>
              <CustomInputBase
                placeholder="Searchâ€¦"
                inputProps={{ 'aria-label': 'search' }}
              />
            </SearchContainer>
          </ToolbarSection>
          <ToolbarSection>
            <ThemeSwitch checked={checked} onChange={handleChange} color="default" />
            <Button color="inherit">Login</Button>
          </ToolbarSection>
        </CustomToolbar>
      </CustomAppBar>
    </>
  );
}

export default Menu;
