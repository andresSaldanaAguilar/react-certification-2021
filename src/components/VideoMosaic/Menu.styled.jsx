import { AppBar, InputBase, Switch, Toolbar } from '@material-ui/core';
import styled from 'styled-components';
import { fade } from '@material-ui/core/styles';

export const CustomAppBar = styled(AppBar)`
  background-color: ${(props) => props.theme.colors.blueGrey} !important;
`;

export const CustomToolbar = styled(Toolbar)`
  display: flex;
  justify-content: space-between;
`;

export const ToolbarSection = styled.div`
  display: flex;
  align-items: center;
`;

export const SearchContainer = styled.div`
  position: relative;
  border-radius: 5px;
  background-color: ${(props) => fade(props.theme.colors.white, 0.15)};
  ${ToolbarSection}:hover & {
    background-color: ${(props) => fade(props.theme.colors.white, 0.25)};
  }
  margin-left: 1em;
`;

export const SearchIconContainer = styled.div`
  padding: 0.5em;
  position: absolute;
  pointer-events: none;
`;

export const CustomInputBase = styled(InputBase)`
  margin: 0.25em;
  padding-left: 2em;
`;

export const ThemeSwitch = styled(Switch)``;
