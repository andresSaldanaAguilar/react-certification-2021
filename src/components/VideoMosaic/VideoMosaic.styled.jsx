import { CardContent } from '@material-ui/core';
import styled from 'styled-components';

export const CustomCardContent = styled(CardContent)`
  background-color: ${(props) => props.theme.colors.blueGrey.light};
  color: white;
  height: 160px;
`;

export const CustomCardContentTitle = styled.div`
  background-color: ${(props) => props.theme.colors.blueGrey.light};
  color: white;
  height: 80px;
`;
