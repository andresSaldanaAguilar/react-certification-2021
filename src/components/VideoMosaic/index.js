export { default } from './VideoMosaic.component';
