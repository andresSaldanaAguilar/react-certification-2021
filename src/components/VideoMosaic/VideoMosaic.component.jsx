import { Card, CardActionArea, CardMedia, Grid, Typography } from '@material-ui/core';
import React from 'react';
import { CustomCardContent } from './VideoMosaic.styled';

function VideoMosaic(props) {
  const { title, thumbnails, description } = props.snippet;
  return (
    <>
      <Grid item xs={4}>
        <Card>
          <CardActionArea>
            <CardMedia
              component="img"
              src={thumbnails.high.url}
              title={title}
              alt={title}
            />
            <CustomCardContent>
              <Typography gutterBottom variant="h5" component="h2">
                {title}
              </Typography>
              <Typography variant="body2" component="p">
                {description}
              </Typography>
            </CustomCardContent>
          </CardActionArea>
        </Card>
      </Grid>
    </>
  );
}

export default VideoMosaic;
