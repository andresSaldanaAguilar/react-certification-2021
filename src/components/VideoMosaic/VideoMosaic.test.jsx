import React from 'react';
import { render, screen } from '@testing-library/react';
import VideoMosaic, { shortenTitle, shortenDescription } from './VideoMosaic.component';
import Theme from '../App/App.styled';
import mockData from '../../utils/youtube-videos-mock-v2.json';

const mockVideo = mockData.items[2];
describe('Video Mosaic Component Tests', () => {
  beforeEach(() => {
    render(
      <Theme>
        <VideoMosaic key={mockVideo.id.videoId} snippet={mockVideo.snippet} />
      </Theme>
    );
  });

  it('Should render the video mosaic', () => {
    expect(screen.getByText(shortenTitle(mockVideo.snippet.title))).toBeInTheDocument();
    expect(
      screen.getByText(shortenDescription(mockVideo.snippet.description))
    ).toBeInTheDocument();
    expect(screen.getByTitle(mockVideo.snippet.title)).toBeInTheDocument();
  });
});
