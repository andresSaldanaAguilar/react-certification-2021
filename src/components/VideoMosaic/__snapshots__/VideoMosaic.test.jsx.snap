// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Mosaic Component Tests Should render the video mosaic 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
  >
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <img
          alt="Video Tour | Welcome to Wizeline Guadalajara"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
          src="https://i.ytimg.com/vi/nmXMgqjQzls/hqdefault.jpg"
          title="Video Tour | Welcome to Wizeline Guadalajara"
        />
        <div
          class="MuiCardContent-root sc-bdfBwQ fUcBBP"
        >
          <h2
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
          >
            Video Tour | Welcome to Wizeline Guadalajara
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Follow Hector Padilla, Wizeline Director of Engineering, for a lively tour of our office. In 2018, Wizeline opened its stunning new office in Guadalajara, Jalisco, ...
          </p>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
